android-easylocation/src/main/java/com/akhgupta/easylocation/EasyLocationDelegate.java

```diff
@@ -210,30 +143,50 @@ void onRequestPermissionsResult(int requestCode, int[] grantResults) {
         switch (requestCode) {
             case PERMISSIONS_REQUEST:
                 if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-                    requestLocation(mLocationRequest, mLocationFetchMode);
                     easyLocationListener.onLocationPermissionGranted();
+                    EasyLocationClientFactory.load(activity, easyLocationClientFactoryLoader);
                 } else
                     easyLocationListener.onLocationPermissionDenied();
                 break;
         }
     }
 
     void onDestroy() {
-        stopLocationUpdates();
-        unregisterLocationBroadcastReceiver();
-    }
-
-    Location getLastKnownLocation() {
-        return PreferenceUtil.getInstance(activity).getLastKnownLocation();
+        EasyLocationClientFactory.stopService(activity);
     }
 
     void requestLocationUpdates(EasyLocationRequest easyLocationRequest) {
-        isProperRequest(easyLocationRequest);
-        requestLocation(easyLocationRequest.locationRequest, CONTINUOUS_LOCATION_UPDATES);
+        if(easyLocationClientFactory == null) {
+            EasyLocationClientFactory.load(activity, easyLocationClientFactoryLoader);
+            return;
+        }
+        easyLocationClient = easyLocationClientFactory
+                .getContinuousUpdatesClient(easyLocationRequest)
+                .listen(new LocationBroadcastReceiver() {
+                    @Override
+                    public void onLocationReceived(Location location) {
+                        easyLocationListener.onLocationReceived(location);
+                    }
+                });
+
     }
 
     void requestSingleLocationFix(EasyLocationRequest easyLocationRequest) {
-        isProperRequest(easyLocationRequest);
```

Comment:
My mistake. I think I meant to move that to the new component. I'll fix that up.